CMakeLists.txt

```diff
@@ -44,16 +52,28 @@ macro(build_pybind11)
   endif()
   list(APPEND extra_cmake_args "-DCMAKE_CXX_FLAGS=${CMAKE_CXX_FLAGS}")
 
-
   include(ExternalProject)
-  ExternalProject_Add(pybind11-2.5.0
-    URL https://github.com/pybind/pybind11/archive/v2.5.0.tar.gz
-    URL_MD5 1ad2c611378fb440e8550a7eb6b31b89
-    TIMEOUT 600
+  ExternalProject_Add(pybind11-2.7.1
+    GIT_REPOSITORY https://github.com/pybind/pybind11.git
```

Comment:
I would recommend using the archive URL over the git one, or at least specifying `GIT_SHALLOW` to avoid downloading all the history of pybind11.

Reply:
Unfortunately, we have to use the `git` URL here.

The problem is the PATCH_COMMAND below.  On Windows, we cannot use just `patch.exe`, since that requires Administrator permission.  That means we have to use `git apply-patch`.  But we've run into issues in the past where `git apply-patch` doesn't do the correct thing unless it is executed inside a git repository.  So we have to use the `GIT_REPOSITORY` here.

I guess I could use `GIT_SHALLOW` here, but we don't do it in the myriad other places we use this pattern.  I'm happy to do a follow-up with all of them if you think that is worthwhile.

Reply:
> I guess I could use GIT_SHALLOW here, but we don't do it in the myriad other places we use this pattern. I'm happy to do a follow-up with all of them if you think that is worthwhile.

Probably not worthwhile on this repo. `time git clone https://github.com/pybind/pybind11.git -b v2.7.1 --depth 1` saves very little time on this repo: 3.6 seconds without `--depth 1` and 3.3 seconds with.